vivisect/base.py

```diff
@@ -832,6 +835,57 @@ def _cb_opcode(self, va, op, branches):
 
         return ()
 
+
+    def _shouldDescend(self, va, arch):
+        '''
+        Check for reasons to skip Function Analysis and Descent
+        This is a overload from envi.codeflow
+
+        This checks for Stop on Map changes and Stop on File changes
+        For File changes, strict mode means any change in filename will cause
+        codeflow to stop.  Non-strict mode means calling from a Filename to
+        an anonymous map (filename == "") is ok, but not the other direction.
+        '''
+        if self._cf_stop_on_exports and self._mem.getExport(va):
+            return False
+
+        if self._cf_onelib:     # Break on change in: map, file, or none
+            mmto = self._mem.getMemoryMap(va)
+            mmfr = self._mem.getMemoryMap(self._cf_calling_va)
+            if mmto is None:
+                logger.info("0x%x->0x%x skipping because destination doesn't exist (no map)", self._cf_calling_va, va)
+                return False
+            if mmfr is None:
+                logger.info("0x%x->0x%x skipping because source doesn't exist (no map)?!?!?", self._cf_calling_va, va)
+                return False
+
+            # if we choose to stop on any map change
+            if self._cf_onelib == envi.CF_RECURSE_STOP_MAPS:
```

Comment:
If I've got two executable memory maps in a binary and `_cf_onelib` to `recurse_stop_maps`, would that mean it wouldn't hop across maps inside of the same file? Is that exepected?

Reply:
yes., it's a granularity of control thing.  it also helps with code executing in anonymous maps that bounce around to other anonymous maps